Detectors/TRD/simulation/src/Detector.cxx

```diff
@@ -121,10 +121,11 @@ bool Detector::ProcessHits(FairVolume* v)
   }
   // Calculate the charge according to GEANT Edep
   // Create a new dEdx hit
-  const double enDep = fMC->Edep();
+  const double enDep = TMath::Max(fMC->Edep(), 0.0) * 1.0e+9;
   // Store those hits with enDep bigger than the ionization potential of the gas mixture for in-flight tracks
   // or store hits of tracks that are entering or exiting
-  if ((enDep > mWion) || trkStat) {
+  int qTot = (int)(enDep / mWion);
+  if (qTot || trkStat) {
```

Comment:
to me the old condition looks the same (and more efficient since you avoid a division). Is this change necessary?

Reply:
I agree with you. I was using this for debugging purposes only.